dts: stm32: Add st,stm32-qdec support to TIM1-TIM5 & TIM8 for STM32F2

AmaanSingh opened the pull request:
QDEC FOR STM32F2: Added the qdec support for the STM32F2 microcontrollers

Currently only the stm32f4 has qdec peripheral in its timers. Other STM32 microcontrollers have qdec but only stm32f4 has support so I added support for the STM32F2 microcontrollers. Will start with other microcontrollers as well. Starts the processing of fixing #88902

Signed-off-by: Amaan Singh <amaansingh160@gmail.com>

AmaanSingh committed:
dts: stm32: Add st,stm32-qdec support to TIM1-TIM5 & TIM8 for STM32F2

Add disabled qdec subnodes to timers TIM1 through TIM5 and TIM8
for STM32F2 series MCUs. This enables Zephyr to provide consistent
QDEC support across all supported encoder-capable STM32 timers.

Signed-off-by: Amaan Singh <amaansingh160@gmail.com>

erwango reviewed (changes requested):
Thanks for this contribution. Additionally to the addition of the nodes, please add a board overlay in samples/sensor/qdec/boards to ensure that the driver could be built and run on such F2 device in CI.

Note:
To do so, it might require to make some changes to sample.yaml file. But maybe removing `sample.sensor.st_qdec_sensor` and relying on `sample.sensor.qdec_sensor` would be sufficient.

AmaanSingh committed:
qdec: Add STM32F2 support via overlay for nucleo_f207zg

Add a disabled st,stm32-qdec node to TIM3 for STM32F2-based MCUs.
This enables QDEC support via devicetree overlays and simplifies
usage on boards such as nucleo_f207zg.

Signed-off-by: Amaan Singh <amaansingh160@gmail.com>

AmaanSingh commented:
Qdec: Add STM32F2 support via overlay for nucleo_f207zg

Added a board overlay in samples/sensor/qdec/boards. Also removed  sample.sensor.st_qdec_sensor and are now relying on sample.sensor.qdec_sensor for the nucleo boards. 

Signed-off-by: Amaan Singh [amaansingh215@gmail.com](mailto:amaansingh215@gmail.com)

AmaanSingh commented:
Qdec: Add STM32F7 support via overlay for nucleo_f767zi

Added a board overlay in samples/sensor/qdec/boards for nucleo_f767zi, and added qdec support for STM32F7 microcontrollers.

Signed-off-by: Amaan Singh [amaansingh215@gmail.com](mailto:amaansingh215@gmail.com)

erwango commented:
@AmaanSingh Before adding more commits, please have a look to compliance issues reported by automatic checks.
Also, we don't take merge commits, so please clean them away from your branch.

AmaanSingh commented:
Should I revert back to the 44e38cf? Because I'm failing compliance checks because of the Merge commits.

AmaanSingh reviewed (commented):
Passed 21 tests and it skipped 3.

erwango commented:
@djiatsaf-st Can you confirm that tests related changes are ok ?

djiatsaf-st commented on samples/sensor/qdec/boards/nucleo_f207zg.overlay:
```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2022 STMicroelectronics
```
Please update Copyright

djiatsaf-st reviewed (commented):
non blocking comment.  Otherwise LGTM

erwango reviewed (changes requested):
Please fix the copyright in original faulty commit.

AmaanSingh commented:
Do you want me to get rid of the copyright and the leading whitespace?

erwango commented:
> Do you want me to get rid of the copyright and the leading whitespace?

Not exactly. The request is that you should not add commits to fix the comments that are done during the review, but fix the initial commits. This is because all your commits will appear in the git tree "as is", so we want clean commits and not a list of commits that fix comments.

AmaanSingh reviewed (commented):
21 passed, 3 skipped
